### ISSUE TITLE
MSSQL generated save() is broken in Propel2

### ISSUE THREAD

**Issue by atheros** [propelorm/Propel2#811]
Cannot save data using MSSQL backend (sqlsrv). 

Fatal error: Call to undefined method Propel\Runtime\ActiveQuery\Criteria::insert() in C:\xxxxxx\model\Base\Group.php on line 1049

``` php
    /**
     * Insert the row in the database.
     *
     * @param      ConnectionInterface $con
     *
     * @throws PropelException
     * @see doSave()
     */
    protected function doInsert(ConnectionInterface $con)
    {
        $this->modifiedColumns[GroupTableMap::COL_ID] = true;
        $criteria = $this->buildCriteria();
        if ($criteria->keyContainsValue(GroupTableMap::COL_ID) ) {
            throw new PropelException('Cannot insert a value for auto-increment primary key (' . GroupTableMap::COL_ID . ')');
        }
        // remove pkey col since this table uses auto-increment and passing a null value for it is not valid
        $criteria->remove(GroupTableMap::COL_ID);
        $pk = $criteria->insert($con);
        $this->setId($pk);  //[IMV] update autoincrement primary key
        $this->setNew(false);
    }
```

Meanwhile the same function using MySQL backend is checking modified keys, generates a prepared statement, binds values and execute the query.

This bug makes MSSQL backend completely unusable with Propel. 

Judging from the difference in code, probably all CRUD operations are affected by similar issue.

As far as I understand, MSSQL lacks integration tests so no one ever knows the status of this backend. In this case, I'd suggest to signal it somehow in the documentation, so people don't lose their time on trying to use this backend.

**Comment by zornado** [comment]
+1 The save function is not working with sqlsrv php 5.6
>Cannot save data using MSSQL backend (sqlsrv).
>This bug makes MSSQL backend completely unusable with Propel.

**Comment by rmarinleal** [comment]
What's happend with this?? I have the same problem migrating a propel2 project from mysql to sqlserver in azure.

**Comment by supermamie** [comment]
On our side, we "fixed" it by adding a new function in Propel/Runtime/activeQuery/Criteria.php : 
    public function insert(ConnectionInterface $con = null)
    {
        return $this->doInsert($con);
    }

**Comment by supermamie** [comment]
Even with this, we are not able to save newly (and empty) created objects.
I will update when we'll manage to make this work.

**Comment by mhaisley** [comment]
I believe this issue was fixed with PR #1239   

@supermamie is it working for you now, or had you given up?   There are a couple of other issues that could still be a problem, and we still need to do a lot to clean up support for Microsoft SQL Server & SQL Azure.

**Comment by supermamie** [comment]
As this issue was found in the very beginning of the project, we managed to use mysql instead so I can not answer :-(
